### PR TITLE
docs: add custom hooks for rendering jupyter notebooks

### PR DESCRIPTION
**Pull request by vbarda**

### CODE REVIEW

File: docs/_scripts/notebook_hooks.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                dest_dir=file.dest_dir,
+                use_directory_urls=file.use_directory_urls,
+            )
+            new_files.append(new_file)
+        else:
+            new_files.append(file)
+    return new_files
+
+
+def on_page_markdown(markdown, page: Page, **kwargs):
```
Comment (eyurtsev) [inline docs/_scripts/notebook_hooks.py:33]: add type hints, what's markdown? 
```suggestion
def on_page_markdown(markdown, page: Page, **kwargs):
```

File: docs/_scripts/notebook_convert.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                # images with the following logic
+                attach_str = f"({name})"
+                if attach_str in cell.source:
+                    data = f"(data:{mime};base64,{data})"
+                    cell.source = cell.source.replace(attach_str, data)
+
+        return cell, resources
+
+
+class CustomRegexRemovePreprocessor(Preprocessor):
```
Comment (eyurtsev) [inline docs/_scripts/notebook_convert.py:106]: If you remember what this does -- would be helpful getting a 1 line summary

**Review by eyurtsev — COMMENTED** [review]
Looks good at a high level, going to test this out

File: docs/_scripts/notebook_convert.py (hunk 2 above)
Comment (vbarda) [reply docs/_scripts/notebook_convert.py:106]: will remove, was cargo culted from langchain script

**Pull request merged**